src/cluster_legacy.c

```diff
@@ -7029,9 +7029,29 @@ int clusterCommandSpecial(client *c) {
         clusterDelNode(n);
         clusterDoBeforeSleep(CLUSTER_TODO_UPDATE_STATE | CLUSTER_TODO_SAVE_CONFIG);
         addReply(c, shared.ok);
-    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && c->argc == 3) {
-        /* CLUSTER REPLICATE <NODE ID> */
+    } else if (!strcasecmp(c->argv[1]->ptr, "replicate") && (c->argc == 3 || c->argc == 4)) {
+        /* CLUSTER REPLICATE (<NODE ID> | NO ONE)*/
         /* Lookup the specified node in our table. */
+        if (c->argc == 4) {
+            if (0 != strcasecmp(c->argv[2]->ptr, "NO") || 0 != strcasecmp(c->argv[3]->ptr, "ONE")) {
```

Comment:
we don't do this 0 != xxx trick, let's keep the same style for the code

Reply:
I see a tons of ```strcasecmp()==0```. You probably want to refactor it then.

Reply:
`strcasecmp()==0` is ok i guess, `0==strcasecmp()` is bad in reading.

Reply:
`0==strcasecmp()` is standard/best practice in C/C++ for some valid reasons. Not a big deal to me.